From c9a86536abfc706c5704533298e71aaaaac56607 Mon Sep 17 00:00:00 2001
From: Rob Clark <rob@ti.com>
Date: Mon, 18 Jan 2010 03:37:36 -0600
Subject: [PATCH 2/4] support to set cropping in sink element

until there is per-buffer meta-data, use caps to inform the sink element of the required cropping of output image
---
 omx/gstomx_base_videodec.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/omx/gstomx_base_videodec.c b/omx/gstomx_base_videodec.c
index 24d51be..7d2fd8c 100644
--- a/omx/gstomx_base_videodec.c
+++ b/omx/gstomx_base_videodec.c
@@ -238,6 +238,11 @@ src_getcaps (GstPad *pad)
                     "width",  G_TYPE_INT, outparam.format.video.nFrameWidth,
                     "height", G_TYPE_INT, outparam.format.video.nFrameHeight,
                     "buffer-count-requested", G_TYPE_INT, outparam.nBufferCountActual,
+                    // XXX hack, crop settings should eventually come as buffer meta-data
+                    "crop-top",  G_TYPE_INT, 24,  // XXX eventually this needs to be figured out dynamically
+                    "crop-left", G_TYPE_INT, 36,  // XXX eventually this needs to be figured out dynamically
+                    "crop-width",  G_TYPE_INT, inparam.format.video.nFrameWidth,
+                    "crop-height", G_TYPE_INT, inparam.format.video.nFrameHeight,
                     NULL);
 
             if(i)
-- 
1.6.3.1

